src/core/stream/representation/representation_stream.ts

```diff
@@ -209,8 +210,22 @@ export default function RepresentationStream<TSegmentDataType>(
     segmentsLoadingCanceller.signal,
   );
 
+  const canStream = new SharedReference<boolean>(true);
+
+  playbackObserver.listen((observation) => {
+    const observationCanStream = observation.canStream ?? true;
+    if (canStream.getValue() !== observationCanStream) {
+      log.debug("Stream: observation.canStream updated to", observationCanStream);
+      canStream.setValue(observationCanStream);
+    }
+  });
+
   /** Emit the last scheduled downloading queue for segments. */
-  const segmentsToLoadRef = segmentQueue.resetForContent(content, hasInitSegment);
+  const segmentsToLoadRef = segmentQueue.resetForContent(
+    content,
+    hasInitSegment,
+    canStream,
```

Comment:
I'm just thinking that now and I don't really care, but we also could have provided that reference higher (e.g. to the `SegmentQueueCreator` or the `createSegmentQueue` method. Any reason why you prefer giving it to `resetForContent`?

Reply:
No particular reason, but yes it feel better to have it as an argument in the `SegmentQueue` constructor.
I removed `ISegmentQueueContentInfo.canLoad` as the value is now stored in `this.canLoad`